src/overlays/actors/ovl_Bg_Hidan_Rock/z_bg_hidan_rock.c

```diff
@@ -47,31 +61,349 @@ static ColliderCylinderInit D_8088BF8C = {
     },
     { 45, 77, -40, { 3310, 120, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Rock/BgHidanRock_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Rock/BgHidanRock_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, -600, ICHAIN_STOP),
+};
+
+void BgHidanRock_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanRock* this = THIS;
+    DynaPolyActor* dyna = &this->dyna;
+    CollisionHeader* col_header = NULL;
```

Comment:
```suggestion
    CollisionHeader* colHeader = NULL;
```
CamelCase is preferred. This will have to change in the rest of the function, too.